src/ReactPlaceholder.js

```diff
@@ -59,15 +57,29 @@ export default class ReactPlaceholder extends React.Component {
     return <Placeholder {...rest} className={classes} />;
   };
 
+  setNotReady = () => {
+    this.timeout = setTimeout(() => {
```

Comment:
`setTimeout(() => {}, 0)` is still asynchrous.

I'm not sure this could cause troubles, but to be sure I would separate the case where delay is `0` from the rest:

```js
const { delay } = this.props;
if (delay > 0) {
  ... your code
} else {
  this.setState({ ready: false });
}
```

Reply:
@FrancescoCioria I agree, not sure if it could cause any issues, but to be on the safe side I bypassed setTimeout in those cases as you suggested. I also added a check to make sure the timeout exists before calling `clearTimeout`.